production.py

```diff
@@ -39,6 +49,140 @@ def wkhtml_to_pdf(cls, data, options=None):
+        dates = [
+            data['start_date'] + relativedelta(days=d) for d in range(
+                (data['end_date'] - data['start_date']).days + 1
+            )
+        ]
+
+        # Dict object to hold productions with key as date and valeue
+        # as list of productions on that day
+        productions_by_date = dict.fromkeys(dates, [])
+
+        # Group searched productions by date and for a dictionary out
+        # of it
+        productions_grouped_by_date = dict(
+            (k, list(v)) for k, v in groupby(
+                sorted(
+                    productions,
+                    key=lambda p: p.effective_date or p.planned_date
+                ),
+                key=lambda p: p.effective_date or p.planned_date
+            )
+        )
+        # Update the values in productions_by_date
+        productions_by_date.update(productions_grouped_by_date)
+
+        localcontext.update({
+            'productions_by_date': productions_by_date,
+            'dates': dates
+        })
+        return super(ProductionsReport, cls).parse(
+            report, records, data, localcontext
+        )
+
+
+class ProductionsReportWizardStart(ModelView):
+    'Generate Productions Report'
+    __name__ = 'report.productions.wizard.start'
+
+    start_date = fields.Date('Start Date', required=True)
+    end_date = fields.Date('End Date', required=True)
+
+    @staticmethod
+    def default_start_date():  # pragma: no cover
+        """
+        Set default start date to the Monday of current week
+        """
+        today = date.today()
+
+        if today.weekday() == 6:
+            # This is to make sure we get monday of current week in case
+            # today is Sunday
+            return today + relativedelta(days=1)
+        return today - relativedelta(days=today.weekday())
+
+    @staticmethod
+    def default_end_date():  # pragma: no cover
+        """
+        Set default end date to the Saturday of current week
+        """
+        today = date.today()
+
+        if today.weekday() == 6:
+            # This is to make sure we get Saturday of current week in case
+            # today is Sunday
+            return today + relativedelta(days=6)
+        return (
+            today - relativedelta(days=today.weekday()) + relativedelta(days=5)
+        )
+
+
+class ProductionsReportWizard(Wizard):
+    'Generate Productions Report Wizard'
+    __name__ = 'report.productions.wizard'
+
+    start = StateView(
+        'report.productions.wizard.start',
+        'production_report.report_productions_view_form', [
+            Button('Cancel', 'end', 'tryton-cancel'),
+            Button('Generate', 'generate', 'tryton-ok', default=True),
+        ]
+    )
+    generate = StateAction(
+        'production_report.report_productions'
+    )
+
+    def do_generate(self, action):
+        """
+        Return report action and the data to pass to it
+        """
+        data = {
```

Comment:
can you move the logic of finding `productions` to here ? This will help in using the schedule report for cases where a few productions are selected from the view. So in short, find the ids of productions and call the report with a list of productions.